xrootdpyfs/fs.py

```diff
@@ -149,8 +149,10 @@ def _raise_status(self, path, status):
         """Raise error based on status."""
         # 3006 - legacy (v4 errno), 17 - POSIX error, 3018 (xrootd v5 errno)
         if status.errno in [3006, 17, 3018]:
+            if status.message.strip().endswith("directory not empty"):
```

Comment:
It might a bit of more work, but why don't we start creating a small mapping/enum `error code: code`?
See [here](https://github.com/xrootd/xrootd/blob/master/src/XProtocol/XProtocol.hh#L1006) and explanation [here](https://github.com/xrootd/xrootd/blob/master/src/XProtocol/XProtocol.hh#L1374).

Reply:
thanks for the suggestion, I added it to our generic issue about improving errors, for now I patched it until we have some breathing room for proper solution
see https://github.com/inveniosoftware/xrootdpyfs/issues/122